chroma_core/lib/graphql.py

```diff
@@ -0,0 +1,27 @@
+import json
+import os
+import requests
+import settings
+from django.db import connection
+
+
+def get_api_session_request():
+    cursor = connection.cursor()
+    cursor.execute("SELECT * from api_key()")
+    (API_USER, API_KEY) = cursor.fetchone()
+    cursor.close()
+    s = requests.Session()
+    s.headers.update({"AUTHORIZATION": "ApiKey {}:{}".format(API_USER, API_KEY)})
+    return s
+
+
+def graphql_query(query):
+    """
+    Issue a GraphQL query
+    """
+    s = get_api_session_request()
+    path = os.path.join(settings.IML_API_PROXY_PASS, "graphql")
```

Comment:
Isn't this going to go to the API port? I thought a call to graphql would be something like this:
```
https://127.0.0.1:8443/graphql/
```
Is this wrong?

Reply:
graphql is handled by `iml-api` and if I go to https, I get TLS errors.

Reply:
Neat, I didn't realize that. I've always made my requests like this:
```
curl -k "https://127.0.0.1:443/graphql/"  
    -X POST
    -d "{ \"query\": \"{ targets { state name activeHostId hostIds filesystems uuid mountPath } }\" }"
    -H "Authorization: ApiKey API:<key>"
    -H "Content-Type: application/json"
```

Reply:
The advantage of going through nginx is that it will provide ApiKey auth (which won't be provided when going directly to the service).

Reply:
I have a feeling this isn't going to work in docker.

Reply:
Inside the job-scheduler container:

```
>>> settings.IML_API_PROXY_PASS
'http://iml-action-runner:8004'
```